### DISCUSSION TITLE
disconnected single chain generated coordinates

### DISCUSSION THREAD

**Discussion by ivanjayapurna** [marrink-lab/polyply_1.0#304] [Q&A] [answered]
Hi, I'm currently struggling to generate a single PMMA chain of an arbitrarily chosen 50 monomers. Both using GROMOS and Martini in PolyPly resulted in what appears to be disconnected monomers instead of one continuous chain. I used the following commands with the martini force field:

polyply gen_params -lib martini3 -o PMMA50_martini.itp -name PMMA -seq PMMA:50
polyply gen_coords -p martini3.top -o martini3_test.gro -name martini3_test -box 5 5 5

Below is martini3.top:

#include "martini_v3.0.0.itp"
#include "PMMA50_martini.itp"
[ system ]
; name
martini3_test
[ molecules ]
; name number
PMMA 1

Vermouth can be used to visualize in 2D and see that the chain is connected:
<img width="487" alt="Screen Shot 2023-03-08 at 3 57 58 PM" src="https://user-images.githubusercontent.com/26446954/223880084-db2d15e6-561b-40b4-a93b-9deb31baed48.png">

however, when visualized in 3D using PyMOL the starting coordinates generated seem entirely disconnected for both forcefields:
<img width="481" alt="Screen Shot 2023-03-08 at 4 00 22 PM" src="https://user-images.githubusercontent.com/26446954/223880312-7aeb34e7-48ff-4212-87c6-9302d102f73d.png">

<img width="466" alt="Screen Shot 2023-03-08 at 4 02 00 PM" src="https://user-images.githubusercontent.com/26446954/223880527-2661057e-e5ee-4acd-beb4-714e63f782ac.png">

Note the commands to generate the latter were:

polyply gen_seq -from_string A:50:1:PMMA-1.0 -o PMMA50_gromos.json -name PMMA -seq A -
label 0:chiral:R-0.5,S-0.5
polyply gen_params -lib 2016H66 -o PMMA50_gromos.itp -name PMMA -seqf
PMMA50_gromos.json
polyply gen_coords -p gromos.top -o gromos.gro -name PMA -box 5 5 5

I received the following errors:
WARNING - general - Node 0 has no resid. Setting resid to 0 + 1.
WARNING - general - Node 1 has no resid. Setting resid to 1 + 1.
WARNING - general - Node 2 has no resid. Setting resid to 2 + 1.
WARNING - general - Node 3 has no resid. Setting resid to 3 + 1.
WARNING - general - Node 4 has no resid. Setting resid to 4 + 1.
WARNING - general - Node 5 has no resid. Setting resid to 5 + 1.
WARNING - general - Node 6 has no resid. Setting resid to 6 + 1.
WARNING - general - Node 7 has no resid. Setting resid to 7 + 1.
WARNING - general - Node 8 has no resid. Setting resid to 8 + 1.
WARNING - general - Node 9 has no resid. Setting resid to 9 + 1.
WARNING - general - Node 10 has no resid. Setting resid to 10 + 1.
WARNING - general - Node 11 has no resid. Setting resid to 11 + 1.
WARNING - general - Node 12 has no resid. Setting resid to 12 + 1.
WARNING - general - Node 13 has no resid. Setting resid to 13 + 1.
WARNING - general - Node 14 has no resid. Setting resid to 14 + 1.
WARNING - general - Node 15 has no resid. Setting resid to 15 + 1.
WARNING - general - Node 16 has no resid. Setting resid to 16 + 1.
WARNING - general - Node 17 has no resid. Setting resid to 17 + 1.
WARNING - general - Node 18 has no resid. Setting resid to 18 + 1.
WARNING - general - Node 19 has no resid. Setting resid to 19 + 1.
WARNING - general - Node 20 has no resid. Setting resid to 20 + 1.
WARNING - general - Node 21 has no resid. Setting resid to 21 + 1.
WARNING - general - Node 22 has no resid. Setting resid to 22 + 1.
WARNING - general - Node 23 has no resid. Setting resid to 23 + 1.
WARNING - general - Node 24 has no resid. Setting resid to 24 + 1.
WARNING - general - Node 25 has no resid. Setting resid to 25 + 1.
WARNING - general - Node 26 has no resid. Setting resid to 26 + 1.
WARNING - general - Node 27 has no resid. Setting resid to 27 + 1.
WARNING - general - Node 28 has no resid. Setting resid to 28 + 1.
WARNING - general - Node 29 has no resid. Setting resid to 29 + 1.
WARNING - general - Node 30 has no resid. Setting resid to 30 + 1.
WARNING - general - Node 31 has no resid. Setting resid to 31 + 1.
WARNING - general - Node 32 has no resid. Setting resid to 32 + 1.
WARNING - general - Node 33 has no resid. Setting resid to 33 + 1.
WARNING - general - Node 34 has no resid. Setting resid to 34 + 1.
WARNING - general - Node 35 has no resid. Setting resid to 35 + 1.
WARNING - general - Node 36 has no resid. Setting resid to 36 + 1.
WARNING - general - Node 37 has no resid. Setting resid to 37 + 1.
WARNING - general - Node 38 has no resid. Setting resid to 38 + 1.
WARNING - general - Node 39 has no resid. Setting resid to 39 + 1.
WARNING - general - Node 40 has no resid. Setting resid to 40 + 1.
WARNING - general - Node 41 has no resid. Setting resid to 41 + 1.
WARNING - general - Node 42 has no resid. Setting resid to 42 + 1.
WARNING - general - Node 43 has no resid. Setting resid to 43 + 1.
WARNING - general - Node 44 has no resid. Setting resid to 44 + 1.
WARNING - general - Node 45 has no resid. Setting resid to 45 + 1.
WARNING - general - Node 46 has no resid. Setting resid to 46 + 1.
WARNING - general - Node 47 has no resid. Setting resid to 47 + 1.
WARNING - general - Node 48 has no resid. Setting resid to 48 + 1.
WARNING - general - Node 49 has no resid. Setting resid to 49 + 1.

Sorry if I missed or misunderstood something extremely simple - any help would be greatly appreciated!

**Comment by fgrunewald** [comment, marked as answer]
@ivanjayapurna thanks for bringing this up. The polymer looks disconnected for two reasons: 1) it goes across the PBC; 2) you always need to run an energy minimisation before the structure becomes connected. Polyply scales the coordinates which upon energy minization unfold so to speak. So try running the energy minization first. If the problem persist we take another look. FYI the warnings are no issue here. They occur because the 'gen_seq' tool currently does not set residue IDs, which is something we want to fix.
